prov/verbs/src/verbs_info.c

```diff
@@ -902,6 +960,8 @@ static int fi_ibv_get_srcaddr_devs(struct fi_info **info)
 	}
 
 	for (fi = *info; fi; fi = fi->next) {
+		if (fi->ep_attr->type == FI_EP_DGRAM)
+			continue;
```

Comment:
Don't you need some sort of src address associated with UD QPs, even if some of the fields will be a wildcard until the actual EP is created?  E.g. couldn't you include the gid or lid?  Hmm... can that idea even work?